db/api/v1/db/assets.db.js

```diff
@@ -22,25 +23,33 @@ const QUERY_FIND = `
  *
  * Returns all assets across all projects unless projectId is given.
  *
+ * @param {number} sponsorId - The ID of the sponsor
  * @param {number} [projectId] - (assumes isValidDbInteger(projectId)): optional: filter assets by this Project ID
+ * @param {number} assetType - the ID of the asset type
+ * @param {Array} donorCodes - a list of donor codes that the asset could have
  * @returns {object[]|undefined} array of assets with fields (id, project_id, latitude, and longitude), or undefined if projectId is invalid.
  * @throws error if the DB query failed to execute
  */
-const find = async (sponsorId, projectId, assetType) => {
+const find = async (sponsorId, projectId, assetType, donorCodes = undefined) => {
     let query = QUERY_FIND;
     let values = [];
-    if ((typeof sponsorId !== 'undefined') & (sponsorId > 0)) {
+    if ((typeof sponsorId !== 'undefined') && (sponsorId > 0)) {
         values.push(sponsorId);
         query = query + `AND sponsor_id = $${values.length}` + ' ';
     }
-    if ((typeof projectId !== 'undefined') & (projectId > 0)) {
+    if ((typeof projectId !== 'undefined') && (projectId > 0)) {
```

Comment:
Formatting: Needs to be brought up to the closing brace

Reply:
This is not an else if

Reply:
Not for a list of `if`s